engine/consensus/compliance/core.go

```diff
@@ -343,7 +343,12 @@ func (c *Core) processBlockProposal(proposal *messages.BlockProposal) error {
 
 	// submit the model to hotstuff for processing
 	log.Info().Msg("forwarding block proposal to hotstuff")
-	c.hotstuff.SubmitProposal(header, parent.View)
+
+	if inRangeBlockResponse {
```

Comment:
```suggestion
	// when the block is in range response, we should wait for hotstuff to finish processing the block,
	// otherwise processing the next block might fail because the current block hasn't been added
	// to protocol state yet.
	if inRangeBlockResponse {
```